rotkehlchen/api/rest.py

```diff
@@ -321,7 +323,10 @@ def _query_async(self, command: Callable, **kwargs: Any) -> Response:
         )
         greenlet.task_id = task_id
         greenlet.link_exception(self._handle_killed_greenlets)
-        self.rotkehlchen.api_task_greenlets.append(greenlet)
+        self.rotkehlchen.api_task_greenlets.append(ApiTaskGreenlet(
```

Comment:
why go through all these? And why make a wrapper with a boolean? This is a terrible approach. Absolutely no. Ask before taking such an approach.

Hacky as fuck with the extra kwargs that is checked in every `query_async` invocation.

Wow ..